src/techui_builder/generate.py

```diff
@@ -315,11 +323,19 @@ def build_groups(self):
         )
 
         self.group.version("2.0.0")
-        self.group.add_widget(widgets)
+        self.group.add_widget(self.widgets)
         self.screen_.add_widget(self.group)
 
     def write_screen(self, directory: Path):
         """Write the screen to file"""
+
+        if self.widgets == []:
+            print(
+                f"Could not write screen: {self.screen_name}\
+                      as no widgets were available"
+            )
```

Comment:
Remove whitespace and add missing 'space
```suggestion
            print(
                f"Could not write screen: {self.screen_name} \
as no widgets were available"
            )
```